File: 4-pointers/Example2.java
public class Example2 {
    public static void process(int x) {
        x = 100;
    }

    public static void main(String[] args) {
        int x = 1;
        process(x);

        System.out.println("x after process(x): " + x); // x == 1 (unchanged)
    }
}

File: 4-pointers/example_2.go
package main

import "fmt"

// Pass by value
func process(x int) {
	x = 100
}

// Pass by reference (pointer)
func processPtr(x *int) {
	*x = 100
}

func main() {
	x := 1
	process(x)

	fmt.Println("x after process(x):", x) // x == 1 (unchanged)

	processPtr(&x)
	fmt.Println("x after processPtr(&x):", x) // x == 100 (changed)
}
